signer/pkcs11/pkcs11_stub.go

```diff
@@ -0,0 +1,24 @@
+// +build !pkcs11
+
+package pkcs11
+
+import (
+	"github.com/cloudflare/cfssl/config"
+	"github.com/cloudflare/cfssl/errors"
+	"github.com/cloudflare/cfssl/signer"
+)
+
+// Config contains configuration information required to use a PKCS
+// #11 key.
+type Config struct {
+	Module string
+	Token  string
+	PIN    string
+	Label  string
+}
+
+// New always returns an error. If PKCS #11 support is needed, the
+// program should be built with the `pkcs11` build tag.
+func New(caCertFile string, policy *config.Signing, cfg *Config) (signer.Signer, error) {
+	return nil, errors.New(errors.PrivateKeyError, errors.Unknown)
+}
```

Comment:
It seems like if you have this level of adaptation, then the split in the CLI code (and associated duplication of non-PKCS#11 code) is unnecessary.  The only function it serves is to hide the PKCS#11 options from the CLI, which could be done more cleanly with something like:

In pkcs11_stub.go:
```
const Enabled = false
```

In pkcs11.go:
```
const Enabled = true
```

In cli/config.go :: registerFlags:
```
if pkcs11.Enabled { /* register p11 flags */ }
```